test/ConfigSpec.hs

```diff
@@ -45,6 +45,13 @@ spec =
           addr = "http://" ++ host ++ ":" ++ port
         in ':' `notElem` port ==> splitAddress addr `shouldBe` (Just "http://", host, port)
       )
+    it "should parse addr without explicit port" $
+      property (\host ->
+        let
+          addr :: String
+          addr = "http://" ++ host
+        in ':' `notElem` host ==> splitAddress addr `shouldBe` (Just "http://", host, "80")
+      )
```

Comment:
Could you add also tests for the other situations that might occur here, namely `https://` and any other string, for example `ftp://`. You can copy the test or you can create a `newtype` wrapper and give an arbitrary instance for that wrapper, like 
```haskell
it "should parse addr without explicit port" $
      property (\protocol host ->
        let
          addr :: String
          addr = unProtocol protocol ++ "://" ++ host
        in ':' `notElem` host ==> splitAddress addr `shouldBe` (Just (unProtocol protocol ++ "://"), host, getPort protocol)
      )
```

but I leave it up to you if you want that.

Reply:
I've added a test for `https://`. Other schemas appear to be rejected, as described by the `should reject invalid schemes` test.